File: FlexServ Backend/SpringBoot/TransactionService/src/main/java/com/flexserv/entity/BookingStatus.java
package com.flexserv.entity;

public enum BookingStatus {
    PENDING,
    REQUESTED,
    CONFIRMED,
    IN_PROGRESS,
    COMPLETED,
    CANCELLED,
    REJECTED;

    public static boolean isValidStatus(String status) {
        if (status == null) return false;
        for (BookingStatus bs : values()) {
            if (bs.name().equalsIgnoreCase(status)) {
                return true;
            }
        }
        return false;
    }
}

File: FlexServ Backend/DotNet/TransactionService/Models/BookingStatus.cs
namespace TransactionService.Models;

public enum BookingStatus
{
    PENDING,
    REQUESTED,
    CONFIRMED,
    IN_PROGRESS,
    COMPLETED,
    CANCELLED,
    REJECTED
}

public static class BookingStatusExtensions
{
    public static bool IsValidStatus(string? status)
    {
        if (string.IsNullOrWhiteSpace(status)) return false;
        return Enum.TryParse<BookingStatus>(status, true, out _);
    }
}
